src/compiler/compile/css/Selector.ts

```diff
@@ -470,10 +470,38 @@ function get_element_parent(node: Element): Element | null {
 	return parent as Element | null;
 }
 
+function find_previous_sibling_considering_slots(node: INode): INode {
+	if (node.component.compile_options.customElement) {
+		return node.prev;
+	}
+
+	let current_node: INode = node;
+	do { 
+		// @ts-ignore
```

Comment:
it'd be nice to add a comment next to each `@ts-ignore` explaining why it's needed